Lib/asyncio/__main__.py

```diff
@@ -101,7 +101,12 @@ def run(self):
                 import tokenize
                 with tokenize.open(startup_path) as f:
                     startup_code = compile(f.read(), startup_path, "exec")
-                    exec(startup_code, console.locals)
+                    try:
+                        exec(startup_code, console.locals)
+                    except SystemExit:
+                        raise
```

Comment:
I *really* don't like this, but it resembles the original behavior of the REPLs -- see GH-143023.
I don't think this is correct, but it's not super clear to me, so please chime in to GH-143023 to put your two cents in.

```suggestion
                    # TODO: Revisit in GH-143023
                    except SystemExit:
                        raise
```

Reply:
GH-143023 is also why I didn't add a test for this.